lib/src/metta/runner/mod.rs

```diff
@@ -115,49 +115,50 @@ impl Metta {
     pub fn run(&self, parser: &mut SExprParser) -> Result<Vec<Vec<Atom>>, String> {
         let mut mode = Mode::ADD;
         let mut results: Vec<Vec<Atom>> = Vec::new();
-
+    
         loop {
-            let atom = parser.parse(&self.tokenizer.borrow());
-            match atom {
-                Some(atom) => {
+            let atom_result = parser.parse(&self.tokenizer.borrow());
```

Comment:
We could use `?` here to re-throw error:
```rs
            let atom_result = parser.parse(&self.tokenizer.borrow())?;
```
Then unpacking result in `match` branches is not needed.